### ISSUE TITLE
Embeddable Factories are no longer supported

### ISSUE THREAD

**Issue by micheh** [zenstruck/foundry#192]
Before #181 it was possible to use a Factory for a Doctrine Embeddable (value object). 

For example (LocationFactory is a factory for an Embeddable):
```php
protected function getDefaults(): array
{
    return [
        'location' => LocationFactory::new(),
    ];
}
``` 

Unfortunately it now checks the provided factory for the object relations:

https://github.com/zenstruck/foundry/blob/39f69f95bb3602d0e77ac5cf8c4359f03c161d95/src/Factory.php#L310

The method `relationshipField` requires the object to be an entity, as it tries to fetch the object manager for the provided argument. Since the object is not an entity, it does not have a manager and a runtime exception is thrown:

https://github.com/zenstruck/foundry/blob/5f39d8a02b0399152a29067af7c1b54ad5514ad6/src/Configuration.php#L157

Is it possible to support factories for Embeddables again?

**Comment by kbond** [comment]
Absolutely, this is a regression. I'll work on a fix.

**Comment by kbond** [comment]
Question: in your `LocationFactory`, are you calling `->withoutPersisting()` in `LocationFactory::initialize()`?

**Comment by micheh** [comment]
Hello @kbond, thank you for looking into this. Correct, here is some code:

```php
final class LocationFactory extends ModelFactory
{
    protected static function getClass(): string
    {
        return Location::class;
    }

    protected function getDefaults(): array
    {
        return [
            'streetAddress' => self::faker()->streetAddress(),
            'postalCode' => self::faker()->postcode(),
            'locality' => self::faker()->city(),
        ];
    }

    protected function initialize(): self
    {
        return $this->withoutPersisting();
    }
}

/**
 * @ORM\Embeddable()
 */
class Location
{
    /**
     * @var string|null
     * @ORM\Column(type="string", nullable=true)
     */
    private $streetAddress;

    /**
     * @var string|null
     * @ORM\Column(type="string", nullable=true)
     */
    private $postalCode;

    /**
     * @var string|null
     * @ORM\Column(type="string", nullable=true)
     */
    private $locality;


    public function __construct(string $streetAddress, string $postalCode, string $locality)
    {
        $this->streetAddress = $streetAddress;
        $this->postalCode = $postalCode;
        $this->locality = $locality;
    }
}
```

**Comment by kbond** [comment]
Perfect, #193 should solve the issue. In addition, calling `->withoutPersisting()` within an embeddable's factory is no longer required!

Are you able to test #193 in your app to ensure no other regressions?

**Comment by micheh** [comment]
This is great, thank you. I tested it with #193 and everything looks good.

**Comment by kbond** [comment]
Great, merged and released in [v1.13.1](https://github.com/zenstruck/foundry/releases/tag/v1.13.1).
